### ISSUE TITLE
BCPL blog: Add ability to pin post.

### ISSUE THREAD

**Issue by danfox01** [baltimorecounty/BCPL-assets#618]
BCPL would like to add "featured" posts that stick to the top of the layout. I am envisioning this would just be a structured content yes/no option. Will include concrete requirements when a resource is available to work on this.

**Comment by tmccoy529** [comment]
@danfox01 Sanjaya and I are starting to look into this now. From what Marty told us it looks like you'll need the following criteria met: 

- add an option in SE for the entry to be selected as pinned. 
- only one item can be pinned at a time. 
- include an image on the entry to show its pinned. 
- potentially add a border or some other type of eye catching decoration 
- if multiple are selected as pinned only pin to the top the entry that is most recently updated

Let us know if this is accurate or what changes if any need to happen.

**Comment by danfox01** [comment]
@tmccoy529 , yes that's perfect. Just three other comments to clarify those criteria

- The image to show an entry is pinned should actually be a font awesome icon. (Maybe [thumbtack ](https://fontawesome.com/v4.7.0/icon/thumb-tack )or [bookmark](https://fontawesome.com/v4.7.0/icon/bookmark)?)

- Just FYI, blog content is also fed into the BCPL homepage. The pinned post should not have an impact on what is displayed on the home page at this time. It might be a future enhancement.

**Comment by martypowell** [comment]
@danfox01 i was envisioning something like this?

https://codepen.io/martypowell/pen/LqWJaw

**Comment by danfox01** [comment]
Perfect.

**Comment by danfox01** [comment]
@sgrg1, what's the status of this one?

**Comment by tmccoy529** [comment]
@danfox01 @sgrg1 sorry Dan, we had some struggles with git functionality and such. Anyway Sanjay is currently trying to get his code compiled so we can get it moved over to SE for testing. Hoping to have this out today by lunch time.

**Comment by danfox01** [comment]
Thanks @tmccoy529. No worries. Thanks for the update.

**Comment by tmccoy529** [comment]
@danfox01 @sgrg1  This has been moved over for testing as well. Sanjay, after I moved it I just did a quick run through to make sure it showed up and I notice that the CSS for the actual pin isn't showing up even though it is on the actual div so some other CSS must be over riding it. Just a heads up.

**Comment by tmccoy529** [comment]
@sgrg1 ....helps to copy over the CSS file......that was my fault.

**Comment by martypowell** [comment]
ok guys so i think the pin needs to move near the top left corner more, maybe something more like this (for positioning)

![image](https://user-images.githubusercontent.com/1933400/52813783-8bb2e580-3068-11e9-9487-cb13df024ee0.png)

@danfox01 @tmccoy529 @sgrg1

**Comment by tmccoy529** [comment]
@martypowell @sgrg1 Sanjay is looking into this. Fighting some real estate issues but we will do a few different options for web services to take a look at and see what they prefer.

**Comment by danfox01** [comment]
If you prefer, @sdurphy can put together a comp. I don't expect you to have to work through design problems too. Unless you want to.

**Comment by martypowell** [comment]
@danfox01 yep, asked sam to mockup over hangouts.

**Comment by danfox01** [comment]
Dammit Marty! Asking other people to do stuff is my one job!

**Comment by danfox01** [comment]
FYI, @sdurphy and I chatted about this a bit. He pointed out that the concept of "pinned" post may suggest that users can actually pin their own posts -- I agree, it's a little confusing.

We decided we should probably call these "featured" posts instead, and tweak the design a little bit. The functionality will be the same, but Sam is going to do a mock up for the design. Stand by...

**Comment by sdurphy** [comment]
@tmccoy529, @martypowell, is this doable?: https://codepen.io/sdurphy/pen/gEOgbP

**Comment by martypowell** [comment]
definitely doable. @danfox01 are you okay with this solution?

**Comment by danfox01** [comment]
Perfect.

**Comment by tmccoy529** [comment]
@danfox01 @martypowell @sgrg1 this is out in dev and ready to be tested.

**Comment by danfox01** [comment]
Looks great! I'll do some quick testing and let you know what I find, then I'll have@jdomasky take a detailed look.

**Comment by sgrg1** [comment]
sounds goood

**Comment by sgrg1** [comment]
comments works on chrome

**Comment by danfox01** [comment]
Looks really good. Just three things from testing:

- IE11 has a responsive issue -- desktop is fine but as you size down, you'll see the label gets wonky.

- We've got a little bit of UX weirdness with the filters. Looks like the featured item changes based on what filter is applied. I would expect the same featured item to always be featured -- regardless of filters.

- Can we shrink the size of the ribbon text slightly? I think 600 weight and 16px size is ideal.

**Comment by martypowell** [comment]
I believe the ie11 responsiveness issue would be resolved if you set the position to relative on the card--blog class

**Comment by tmccoy529** [comment]
@danfox01 Hey Dan, I was just playing with the filters and I had a question (might add some complexity) but would we want each category filter to allow its own featured blog entry? Or do you just want the one featured blog to show in all selections?

**Comment by martypowell** [comment]
Just one featured entry regardless of category.

**Comment by sgrg1** [comment]
@tmccoy529 fixed issues from previous testing. ready for testing.

**Comment by tmccoy529** [comment]
@danfox01 I moved this code over to SE so this is ready for re-test. (when you get back that is).

**Comment by martypowell** [comment]
@tmccoy529 @sgrg1 can you guys please get this back up in dev?

**Comment by danfox01** [comment]
@tmccoy529, this looks great. One thing after a quick test: The ribbon still has responsive positioning issues in IE11. Here was Marty's comment from the last time we worked on this:

> Believe the ie11 responsiveness issue would be resolved if you set the position to relative on the card--blog class. Right now, I think it's on the body and causing the ribbon to float to the top.

I'll look more thoroughly tomorrow.

**Comment by martypowell** [comment]
@danfox01 yes, I `firebugged` that and it worked.

**Comment by danfox01** [comment]
Just spent 10 min looking for a synonym for this. Doesn't exist. Firebugging it is.

**Comment by tmccoy529** [comment]
@danfox01 Sanjay added that css option and Ive moved the code over to SE. This is ready for retest.

**Comment by martypowell** [comment]
strictly from a technical perspective this is fixed. @danfox01 mobile card is a little funky, is that how it's always looked?

**Comment by danfox01** [comment]
@martypowell, can you define funky? it probably always has been like this but if you have suggestions we can revisit.

**Comment by martypowell** [comment]
@danfox01 they layout just looks like maybe it was never fully finished, but maybe that the is way it has always looked?

**Comment by martypowell** [comment]
@danfox01 moving to ready for release if we want to address the styles we can in another issue.

**Comment by tmccoy529** [comment]
@danfox01 @martypowell do we have a status on this? Last I heard this was complete. Can this be moved back into test or are we missing something still? I'll have to rebuild this again in test so let me know when that should be done.

**Comment by danfox01** [comment]
I don't know. This was in ready for release for a while, but I looks like @martypowell moved it back.

**Comment by martypowell** [comment]
I believe I moved this back because it wasn't available in dev after a refresh. My bad for not updating. We can discuss in our meeting today.

**Comment by martypowell** [comment]
@tmccoy529 looks like maybe there is something up with the service?

**Comment by tmccoy529** [comment]
@martypowell SE didn't have the field name for the blog of IsFeatured. When I added that back into SE and flagged a blog with True I'm seeing the featured ribbon again. Is that was you were referring to?

**Comment by martypowell** [comment]
@tmccoy529 yep makes sense, probably should have thought of that. Thanks for adding it back. @danfox01 is there anything (apart from a change control) preventing us from moving this to production?

**Comment by tmccoy529** [comment]
@danfox01 @martypowell Ive moved this back over to dev again after this last refresh. Partially because without it dev was blowing up in the blog section but also this gives us almost 2 weeks before it goes bye bye from a refresh in case we want to test one last time before this gets pushed out.

**Comment by danfox01** [comment]
@jdomasky can you please test this again? Let me know if you need details on expected behavior.

**Comment by jdomasky** [comment]
@danfox01 No problem. I'll test on Friday and let you know if I need more details.

**Comment by jdomasky** [comment]
I completed testing in dev and the functionality works as expected in authoring and rendering.

However, the IE11 responsiveness problem is occurring: at tablet width and below, the ribbon jumps to the top right corner of the browser window. This problem also occurs in the Edge browser.

If this problem can be resolved, I believe this enhancement is ready for production.

**Comment by danfox01** [comment]
@tmccoy529 @jdomasky What's up with the "IsFeatured" appearing twice in the content collection?
![FireShot Capture 060 - View_ #BWELLREAD in 2019 with WBAL-TV - auth dev baltimorecountymd gov](https://user-images.githubusercontent.com/13049389/56838942-2f312b00-684e-11e9-8c85-f3007c33e771.png)

**Comment by tmccoy529** [comment]
@danfox01 Sorry about that. When I created this again it added a space at the end of the column name. Well a Space changed the name of the column for the web service and broke it. Unfortunately you cannot delete a field in SE after its been used so I had to add another field with the correct name. Its just the hazards of SE I guess. I added a comment to the actual field to ignore it but that is the best I can do.

**Comment by tmccoy529** [comment]
@jdomasky @danfox01 @martypowell well I screwed that build up Jason. We had fixed that responsiveness before and I pushed over an old copy of the blog css file. I've pushed over the correct version and it appears to be working now. Can you do a quick run through again? Im assuming this is how its supposed to look?

![image](https://user-images.githubusercontent.com/40430541/56895382-fadc8b00-6a56-11e9-97ea-36d4a6261f56.png)

**Comment by tmccoy529** [comment]
@jdomasky @danfox01 @martypowell  as I'm testing I see that Chrome likes the changes only. IE doesn't work so back to development again. I'll let you know when its ready to test again.

**Comment by tmccoy529** [comment]
@jdomasky @danfox01 @martypowell okay sorry. There was a lot of confusion because this fix was actually part of the BCPL -Assets project and not part of the react-blog project. This got lost along the way. The new master page for bcpl has been pushed over as well as the correct version of the react-blog, updated SE with the field and pushed out the web service. This is READY TO BE TESTED for real this time. Sorry for the confusion.

**Comment by jdomasky** [comment]
> 
> 
> @tmccoy529 @jdomasky What's up with the "IsFeatured" appearing twice in the content collection?

@danfox01 I didn't mention the duplication of IsFeatured in my testing feedback because the tooltip implied this was an artifact of development. This was reinforced when I discovered that the duplicate setting had no effect.

**Comment by jdomasky** [comment]
@danfox01 @tmccoy529 I should be able to retest this issue soon.

**Comment by jdomasky** [comment]
@danfox01 @tmccoy529 The retest was a success!

**Confirmations:**
- Works great in IE/Edge.
- I confirmed on mobile in BrowserStack.
- I featured two posts and only saw the more recently updated post. When I updated the one I didn't see, it took over the featured spot.
- The featured posts had no impact on what is displayed on the dev home page.
- The most recently updated featured post is displayed regardless of the category filter.
